ERCS/erc-7595.md

```diff
@@ -0,0 +1,491 @@
+	 *
+	 * Requirements:
+	 * - only owner of NFT
+	 *
+	 * @param address collection address.
+	 * @param tokenId unique identifier for specific NFT.
+	 * @param amounts[] array of amounts to collateralize.
+	 * @param address[] array of token addresses.
+	 */
+	function uncollateralize(
+		address collection,
+		uint256 tokenId,
+		uint256[] memory amounts,
+		address[] memory tokenAddresses
+	) external;
+	
+	/**
+	 * @dev Split collateral among all existent tokens.
+	 *
+	 * @param address collection address.
+	 * @param amounts[] to be dispersed among all NFT owners.
+	 * @param address[] address of token to be dispersed.
+	 */
+	function disperse(
+		address collection,
+		uint256[] memory amounts,
+		address[] memory tokenAddresses
+	) external payable;
+}
+```
+
+## Rationale
+### “Envious” Term Choice
+We propose adopting the term "Envious" to describe any NFT collection minted using this ERC standard or any [ERC-721](./eip-721.md) based NFT collection that utilized the EnviousHouse abstraction layer.
+
+### NFT Collateralization with Multiple Tokens
+Some Web3 projects primarily collateralize a specific NFT asset with one [ERC-20](./eip-20.md) based token, resulting in increased gas fees and complications in User Experience (UX).
+
+This ERC has been crafted to enable the collateralization of a designated NFT asset with multiple [ERC-20](./eip-20.md) based tokens within a single transaction.
+
+### NFT Collateralization with the Native Coin
+Each [ERC-20](./eip-20.md) based token possesses a distinct address. However, a native coin does not carry an address. To address this, we propose utilizing a null address (`0x0000000000000000000000000000000000000000`) as an identifier for the native coin during collateralization, as it eliminates the possibility of collisions with smart contract addresses.
+
+### Disperse Functionality
+We have implemented the capability to collateralize all assets within a particular NFT collection in a single transaction. The complete collateral amount is deposited into a smart contract, enabling each user to claim their respective share of the collateral when they add or redeem collateral for that specific asset.
+
+### Harvest Functionality
+Each Envious NFT collection provides an option to incorporate a community [ERC-20](./eip-20.md) based token, which can be exchanged for commissions accrued from collateralization and uncollateralization activities.
+
+### BlackHole Instance
+Some [ERC-20](./eip-20.md) based token implementations forbid transfers to the null address, it is necessary to have a reliable burning mechanism in the harvest transactions. blackHole smart contract removes [ERC-20](./eip-20.md) communityTokens from the circulating supply in exchange for commission fees withdrawn.
+
+blackHole has been designed to prevent the transfer of any tokens from itself and can only perform read operations. It is intended to be used with the Envious extension in implementations related to commission harvesting.
```

Comment:
```suggestion
`blackHole` has been designed to prevent the transfer of any tokens from itself and can only perform read operations. It is intended to be used with the Envious extension in implementations related to commission harvesting.
```

use backticks for any inline code.

Reply:
backticks added